$NetBSD: patch-an,v 1.3 2007/01/20 18:55:09 wiz Exp $

https://bugzilla.mozilla.org/show_bug.cgi?id=323977
https://bugzilla.mozilla.org/attachment.cgi?id=209007&action=view

Added DragonFly.

Also, when building on solaris with gcc, be sure to correctly set the
flags used by the solaris assembler which is explicitly called out in
the makefile in a couple of places.  This prevents errors when trying to
assemble files which contain cpu specific instructions.

--- mozilla/security/nss/lib/freebl/Makefile.orig	2006-08-23 22:57:26.000000000 +0000
+++ mozilla/security/nss/lib/freebl/Makefile
@@ -196,6 +196,17 @@ ifeq (,$(filter-out BSD_OS FreeBSD Linux
     MKSHLIB += -Wl,-Bsymbolic
 endif
 
+# The blapi functions are defined not only in the freebl shared
+# libraries but also in the shared libraries linked with loader.c
+# (libsoftokn3.so and libssl3.so).  We need to use GNU ld's
+# -Bsymbolic option or the equivalent option for other linkers
+# to bind the blapi function references in FREEBLVector vector
+# (ldvector.c) to the blapi functions defined in the freebl
+# shared libraries.
+ifeq (,$(filter-out BSD_OS FreeBSD Linux NetBSD DragonFly, $(OS_TARGET)))
+    MKSHLIB += -Wl,-Bsymbolic
+endif
+
 ifeq ($(OS_TARGET),SunOS)
 
 # The -R '$ORIGIN' linker option instructs this library to search for its
